### DISCUSSION TITLE
How to restore?    or- Backup / Restore Docs?

### DISCUSSION THREAD

**Discussion by MeanderingWanderer** [Koenkk/zigbee2mqtt#14586] [General]
After trying a few different attempts of restoring the info from my backup.json, (There are no docs available on zigbee2mqtt.io)
I repaired 41 devices--rather then spend unknown time searching the internet for the info..

Can we start some sort of how-to restore thread?   or a starting point for user docs?

I'm hardly proficient in Z2M, and don't know where to start.
Thanks!

**Comment by Koenkk** [comment]
The backup is everything inside the `data` directory. So if you want to move to e.g. a different system simply move the `data` folder.

**Reply by deltabert** [reply]
Please @Koenkk would you please look at my issue here: https://github.com/Koenkk/zigbee2mqtt/discussions/31409
It would be much appreciated.
Kind regards Bert

**Reply by Crocmagnon** [reply]
Would it be possible to integrate this with HA backups when run as a HA app? So that the relevant `data` directory is backed up and restored? This would also help in disaster recovery situations.

**Comment by TorbenMei** [comment]
Hi
I have a little doubt in this matter and I am not particularly strong in what happens outside the "Frontend" - which is simply fantastic.
1. How to find the "data" folder?
2. Is it correctly understood that data is not stored on the coordinator (in my case Sonoff Zigbee 3.0 USB Dongle Plus) when the system is shut down. Can you just put another corresponding coordinator in at restart and then the system runs again?
Thanx in advance.
Torben

**Comment by rct** [comment]
I just stumbled across this and came to report an issue, but decided to check here first. 

Backing up and restoring the Home Assistant Addon for Zigbee2MQTT does very little. Most of the addon's relevant data is in the Home Assistant configuration directory(*) and is (hopefully) backed up with Home Assistant. 

I find this to be somewhat surprising/counter-intuitive.

(*) The Home Assistant Configuration directory is the one where `configuration.yaml` is stored.  `/config` on HaOS. I think Home Assistant) is now calling that "_the data directory_". 

Backing up the add-on only backups two small json files:

```
$ tar tvf 45df7312_zigbee2mqtt.tar.gz
drwx------ root/root         0 2023-05-08 11:54 ./
-rw------- root/root      3606 2023-05-08 11:54 ./addon.json
drwxr-xr-x root/root         0 2023-05-07 15:34 data/
-rw------- root/root       404 2023-05-07 15:34 data/options.json
```

Those only contain some minimal configuration for the add-on itself like the path to the serial port, port numbers, MQTT connection, etc.

The data about the Zigbee devices, the coordinator backup, etc. are all part of your Home Assistant backup.  So if you do want to migrate that to a new host, you need to pull that out of the previous Home Assistant instance's backup. 

@Koenkk wrote above:
> The backup is everything inside the `data` directory. So if you want to move to e.g. a different system simply move the `data` folder.

So data directory in this case means what people may think of as the Home Assistant configuration directory and or "Home Assistant Backup".   While it is true that is what you want to move, this statement is a little incomplete.  

If you are migrating the whole Home Assistant installation, things will "just work" when you restore the backup.   Otherwise you need to do some work to copy out/over the Zigbee2mqtt config. 

I think all of this should be a bit more clearly documented.

**Comment by Tokn59** [comment]
I made a backup of my zigbee2mqtt system using the backup option in Settings - Tools - Request z2mbackup. This produces a ZIP file with the following content:

- configuration.yaml
- database.db
- devices.yaml
- groups.yaml
- state.json

According to docs I simply need to copy the database.db to my new zigbee2mqtt installation (in my case Zigbee2mqtt running in a lxc container on Proxmox) This does not seem to work. Zigbee2mqtt starts but there are no devices. Copying the entire set of files does not seem to work either.

What is the correct procedure please ? I am trying to migrate from Zigbee2mqtt as an add-on in HA using a Conbee II stick to a "standalone" Zigbee2mqtt installation with a Sonoff 3 dongle. THANK YOU !

**Reply by roastcom** [reply]
I have a strong suspicion that is due to a change in coordinators. The DB is probably tied to it and since you're no longer using the old one nothing is pulled up.

**Comment by zajaaczki** [comment]
Hi I have different question I need to update my sonoff cc2652p. I have done backup same as Tokn59 5 files. Now I wanna update koenkk firmware using Texas Instruments Flasher v2 and back again dongle to my ha. My question is: The same dongle after update start to work without any issue or I have to restore backup? I am not sure all data are stored on the z2m side or something is also inside dongle firmware?

**Comment by Syridian** [comment]
> According to docs I simply need to copy the database.db to my new zigbee2mqtt installation (in my case Zigbee2mqtt running in a lxc container on Proxmox) This does not seem to work. Zigbee2mqtt starts but there are no devices. Copying the entire set of files does not seem to work either.

From what I can tell you need to copy/restore all of the files to the data folder, not just the database.db file.  I need to test this myself, but I'm not currently in a position where I have a test setup I can do this with.  It would be good to have some sort of confirmation from someone that this indeed does restore all of the devices and the full Zigbee mesh network, or whether I need to then reconnect all of the devices manually.

**Comment by Hedda** [comment]
> Can we start some sort of how-to restore thread? or a starting point for user docs?

If you do then also check out this existing related discussion for refernce -> https://github.com/Koenkk/zigbee2mqtt/discussions/16478

**Comment by 0verEngineer** [comment]
@Koenkk Can you please explain how this works if i want to backup and restore to a new adapter?

**Reply by Hedda** [reply]
See the FAQ for adapters based on Texas Instruments radio SoC series:

https://www.zigbee2mqtt.io/guide/faq/#how-do-i-migrate-from-one-adapter-to-another

https://www.zigbee2mqtt.io/guide/faq/#how-do-i-move-my-zigbee2mqtt-instance-to-a-different-environment

For adapters based on other SoCs see this previously linked discussion:

https://github.com/Koenkk/zigbee2mqtt/discussions/16478

**Comment by Tonio16** [comment]
Hello

It would be nice to have the possibility to upload the backup to restore zigbee2mqtt from the frontend. I want to move from a raspberry Pi to a LXC container, and I have no idea how I can upload the backup in the LXC...

A possible improvement ?

Antoine

**Comment by andi-at-1** [comment]
Maybe someone could update / clarify this topic somewhere, i found no actual working post:

The path (at least in my case) is homeassistant/zigbee2mqtt/
You get there by install ssh addon, enable sftp (and set user to root), then on ubuntu type ssh://root@yourHAip, on windows get something like winscp

Alternative i think the samba addon will work too

If you want to restore to another HA instance you have to

(0) Download config from the old HA instance by Z2M-Config-Tools-request z2m backup (maybe translates different...)
1) Install z2m on the new  HA instance
2) parse the backed up files in the homeassistant/zigbee2mqtt  directory, overwrite the old ones, DONT START z2m!!!
3) !! reload / restart HA, otherwise the just copied config won´t be read !!
4) start z2m, be happy with your old devices

**Reply by sunib** [reply]
This worked for me as well, thanks!

**Comment by Tonio16** [comment]
> Maybe someone could update / clarify this topic somewhere, i found no actual working post:
> 
> The path (at least in my case) is homeassistant/zigbee2mqtt/ You get there by install ssh addon, enable sftp (and set user to root), then on ubuntu type ssh://root@yourHAip, on windows get something like winscp
> 
> Alternative i think the samba addon will work too
> 
> If you want to restore to another HA instance you have to
> 
> (0) Download config from the old HA instance by Z2M-Config-Tools-request z2m backup (maybe translates different...)
> 
> 1. Install z2m on the new  HA instance
> 2. parse the backed up files in the homeassistant/zigbee2mqtt  directory, overwrite the old ones, DONT START z2m!!!
> 3. !! reload / restart HA, otherwise the just copied config won´t be read !!
> 4. start z2m, be happy with your old devices

Tx

Antoine
